### PR TITLE
[DNM] Enable ironic testing in component lines

### PR DESCRIPTION
**Pull request by pablintino**

### CODE REVIEW

File: zuul.d/edpm_multinode.yaml [hunk 1]
```diff
@@ -445,6 +445,33 @@
     parent: podified-multinode-edpm-deployment-crc-bootstrap-staging
     vars:
       cifmw_enable_virtual_baremetal_support: true
+      cifmw_config_nmstate: true
+      cifmw_ci_nmstate_instance_config:
+        controller:
+          interfaces:
+            - name: "{{ cifmw_networking_env_definition.instances['controller'].networks['baremetal'].interface_name | default(omit) }}"
```
Comment (hjensas) [inline zuul.d/edpm_multinode.yaml:452]: Based on the error in CI, I guess the `default(omit)` ends up being used as the name, either here or on L464?

**Comment by pablintino** [comment]
recheck

File: zuul.d/edpm_multinode.yaml (hunk 1 above)
Comment (pablintino) [reply zuul.d/edpm_multinode.yaml:452]: Fixed, we weren't loading the cifmw_networking_env_definition in the deploy-edpm playbook

**Comment by hjensas** [comment]
If possible we should avoid setting mulitple default routes?

```
2024-10-16 14:04:32.514126 | controller |         routes:
2024-10-16 14:04:32.514134 | controller |           config:
2024-10-16 14:04:32.514142 | controller |           - destination: 0.0.0.0/0
2024-10-16 14:04:32.514151 | controller |             next-hop-address: 192.168.122.1
2024-10-16 14:04:32.514175 | controller |             next-hop-interface: ens8
2024-10-16 14:04:32.514187 | controller |           - destination: 0.0.0.0/0
2024-10-16 14:04:32.514195 | controller |             next-hop-address: 172.17.0.1
2024-10-16 14:04:32.514204 | controller |             next-hop-interface: ens8.20
2024-10-16 14:04:32.514212 | controller |           - destination: 0.0.0.0/0
2024-10-16 14:04:32.514221 | controller |             next-hop-address: 172.18.0.1
2024-10-16 14:04:32.514229 | controller |             next-hop-interface: ens8.21
2024-10-16 14:04:32.514241 | controller |           - destination: 0.0.0.0/0
2024-10-16 14:04:32.514250 | controller |             next-hop-address: 172.19.0.1
2024-10-16 14:04:32.514259 | controller |             next-hop-interface: ens8.22
```

**Comment by hjensas** [comment]
`NmstateError: VerificationError: Failed to apply DNS config: desire name servers '', got '127.0.0.1'\n'"`

```
  desiredState:
    dns-resolver:
      config:
        search:
        - default.example.com
        - internal-api.example.com
        - storage.example.com
        - tenant.example.com
        server: []
```

File: zuul.d/edpm_multinode.yaml [hunk 2]
```diff
@@ -445,11 +442,37 @@
     parent: podified-multinode-edpm-deployment-crc-bootstrap-staging
     vars:
       cifmw_enable_virtual_baremetal_support: true
+      cifmw_config_nmstate: true
```
Comment (hjensas) [inline zuul.d/edpm_multinode.yaml:445]: I can't grok from where the NNCP for the CRC is built?
But looking at https://logserver.rdoproject.org/53/2453/620324a4416ac7f15d6a28a9178c7807ba2617e0/github-check/podified-multinode-edpm-deployment-crc-bootstrap-staging-ironic/06383e3/controller/ci-framework-data/artifacts/manifests/nmstate/crc-nncp.yaml - it does not have the additional baremetal interface.

Comment (pablintino) [reply zuul.d/edpm_multinode.yaml:445]: That part is still missing, all I'm trying to do get the nmstate part to work. Give me some more patches and I'll reach that part.

**Comment by pablintino** [comment]
recheck

**Comment by pablintino** [comment]
recheck

**Comment by pablintino** [comment]
recheck

**Comment by lewisdenny** [comment]
@pablintino I believe this one can be closed now all the changes have been taken out and placed in smaller PRs.

**Pull request closed without merging**